addons/l10n_tw_edi_ecpay/models/account_move.py

```diff
@@ -471,23 +471,24 @@ def _l10n_tw_edi_prepare_item_list(self, json_data, is_allowance=False):
 
             twd_excluded_amount = base_line['tax_details']['raw_total_excluded']
             twd_included_amount = base_line['tax_details']['raw_total_included']
+            quantity = (-1 if line.quantity < 0 else 1) * line.quantity
```

Comment:
```suggestion
            quantity = abs(line.quantity)
```
Having a comment here explaining why this is needed would also be nice ☺️

Reply:
Since in the task it's mentioned that "ECPay ... does not seem to accept -ve quantity", this might be necessary for other cases where the quantity is negative :thinking: Maybe @kidoodoo can comment on this :)

Reply:
ECPay (The e-invoicing platform) does not accept negative quantity.

On my side, did not find a case other than downpayment that Odoo uses negative quantity as part of its own behaviour.

The most similar case I could think of was the global discount applied from sales order coming to invoice but it has positive quantity, with negative price which is accepted.

So in terms of the requirement side, both should work whether we fixate this fix to downpayment or every time there is a negative quantity since I cannot think of other flow on default that would trigger negative quantity by default.

Doing it the way at the moment would be future-proof in case there is another flow with negative quantity or new flow is introduced or a user decides to do it for their own reason I believe.

Will think a bit more to see if there is any other feature in Odoo trigger negative quantity on invoice

Reply:
OK, then let's keep the fix as a generic approach for now.
We can always adapt the code later if the need arises.

Reply:
Does it accept negative prices?
Shouldn't we put the sign on the price instead of the quantity?

Just using the absolute value is wrong in almost every single case

Reply:
In simplest sense without going into details, ItemPrice is unit price and ItemAmount is ItemPrice * ItemCount(Quantity)

https://developers.ecpay.com.tw/?p=24230 (B2B invoice issuance documentation)

Example JSON from their developer documentation for item line.
        {
            "ItemSeq": 3,
            "ItemName": "item03",
            "ItemCount": 3,
            "ItemWord": "粒",
            "ItemPrice": 10,
            "ItemAmount": 30,
            "ItemTax": 2,
            "ItemRemark": "item03_desc"
        }